### PR TITLE
Refactor MyTardis to support MySQL deployment

### PR DESCRIPTION
**Pull request by nrmay**
This PR includes changes required to support the deployment and testing of MyTardis on MySQL.
Including:

1. Truncate URL, Char, and indexed fields to max. 255. Models affected include:
  * datafile
  * dataset
  * experiment
  * fields
  * license
  * storage

2. Migration files (new and squashed) to support the model changes.
3. Add truncation of microseconds in date utilities —as this is not supported uniformly by different RDBMS.  

4. Refactor tests to support transactions in MySQL, i.e. switch inheritance from TestCase to TransactionTestCase.  

5. Add filename as parameter in testcase (test_rmexperiment.py).  

6. Add pylint disables in test_on_mysql_settings.py.

Tested on MySQL (5.6.30) and PostgreSQL (9.4.1). see: [nrmay on semaphore](https://semaphoreci.com/nrmay)

### CODE REVIEW

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+        df.save()
+        current_object += 1
+        if current_object % 10000 == 0:
+            print "{0} of {1} datafile objects converted".format(
+                    current_object, total_objects)
+
+
+class Migration(migrations.Migration):
+
+    replaces = [(b'tardis_portal', '0001_initial'), (b'tardis_portal', '0002_auto_20150528_1128'), (b'tardis_portal', '0003_auto_20150907_1315'), (b'tardis_portal', '0004_storageboxoption_value_type'), (b'tardis_portal', '0005_datafile_add_size_int_column'), (b'tardis_portal', '0006_datafile_remove_size_string_column'), (b'tardis_portal', '0007_remove_parameter_string_value_index'), (b'tardis_portal', '0008_string_value_partial_index_postgres'), (b'tardis_portal', '0009_auto_20160128_1119'), (b'tardis_portal', '0010_auto_20160503_1443'), (b'tardis_portal', '0011_auto_20160505_1643')]
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (590/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l33)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+        migrations.CreateModel(
+            name='ParameterName',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('name', models.CharField(max_length=60)),
+                ('full_name', models.CharField(max_length=60)),
+                ('units', models.CharField(max_length=60, blank=True)),
+                ('data_type', models.IntegerField(default=2, choices=[(1, b'NUMERIC'), (2, b'STRING'), (3, b'URL'), (4, b'LINK'), (5, b'FILENAME'), (6, b'DATETIME'), (7, b'LONGSTRING')])),
+                ('immutable', models.BooleanField(default=False)),
+                ('comparison_type', models.IntegerField(default=1, choices=[(1, b'Exact value'), (8, b'Contains'), (3, b'Range'), (4, b'Greater than'), (5, b'Greater than or equal'), (6, b'Less than'), (7, b'Less than or equal')])),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (232/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l313)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+            unique_together=set([('datafile', 'storage_box')]),
+        ),
+        migrations.AlterUniqueTogether(
+            name='datafile',
+            unique_together=set([('dataset', 'directory', 'filename', 'version')]),
+        ),
+        migrations.AlterField(
+            model_name='parametername',
+            name='data_type',
+            field=models.IntegerField(default=2, choices=[(1, b'NUMERIC'), (2, b'STRING'), (3, b'URL'), (4, b'LINK'), (5, b'FILENAME'), (6, b'DATETIME'), (7, b'LONGSTRING'), (8, b'JSON')]),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (189/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l534)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+        ),
+        migrations.AlterField(
+            model_name='datafile',
+            name='mimetype',
+            field=models.CharField(db_index=True, max_length=80, blank=True),
+        ),
+        migrations.AlterField(
+            model_name='datafile',
+            name='directory',
+            field=tardis.tardis_portal.models.fields.DirectoryField(max_length=255, null=True, blank=True),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (107/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l568)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 5]
```diff
@@ -0,0 +1,600 @@
...
+        migrations.CreateModel(
+            name='License',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('name', models.CharField(unique=True, max_length=400)),
+                ('url', models.URLField(help_text=b'Link to document outlining licensing details.', unique=True, max_length=255)),
+                ('internal_description', models.TextField()),
+                ('image_url', models.URLField(max_length=255, blank=True)),
+                ('allows_distribution', models.BooleanField(default=False, help_text=b'Does this license provide distribution rights?')),
+                ('is_active', models.BooleanField(default=True, help_text=b'Can experiments continue to select this license?')),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (128/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l280)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+        ('contenttypes', '0002_remove_content_type_name'),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('auth', '0006_require_contenttypes_0002'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='DataFile',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (114/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l45)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 7]
```diff
@@ -0,0 +1,600 @@
...
+        ),
+        migrations.CreateModel(
+            name='DatafileParameter',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('string_value', models.TextField(null=True, blank=True)),
+                ('numerical_value', models.FloatField(db_index=True, null=True, blank=True)),
+                ('datetime_value', models.DateTimeField(db_index=True, null=True, blank=True)),
+                ('link_id', models.PositiveIntegerField(null=True, blank=True)),
+                ('link_ct', models.ForeignKey(blank=True, to='contenttypes.ContentType', null=True)),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (101/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l81)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 8]
```diff
@@ -0,0 +1,600 @@
...
+                ('pluginId', models.CharField(max_length=30)),
+                ('entityId', models.CharField(max_length=320)),
+                ('object_id', models.PositiveIntegerField()),
+                ('canRead', models.BooleanField(default=False)),
+                ('canWrite', models.BooleanField(default=False)),
+                ('canDelete', models.BooleanField(default=False)),
+                ('isOwner', models.BooleanField(default=False)),
+                ('effectiveDate', models.DateField(null=True, blank=True)),
+                ('expiryDate', models.DateField(null=True, blank=True)),
+                ('aclOwnershipType', models.IntegerField(default=1, choices=[(1, b'Owner-owned'), (2, b'System-owned')])),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (122/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l296)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 9]
```diff
@@ -0,0 +1,600 @@
...
+                ('title', models.CharField(max_length=400)),
+                ('institution_name', models.CharField(default=b'Monash University', max_length=400)),
+                ('description', models.TextField(blank=True)),
+                ('start_time', models.DateTimeField(null=True, blank=True)),
+                ('end_time', models.DateTimeField(null=True, blank=True)),
+                ('created_time', models.DateTimeField(auto_now_add=True)),
+                ('update_time', models.DateTimeField(auto_now=True)),
+                ('handle', models.TextField(null=True, blank=True)),
+                ('locked', models.BooleanField(default=False)),
+                ('public_access', models.PositiveSmallIntegerField(default=1, choices=[(1, b'No public access (hidden)'), (25, b'Ready to be released pending embargo expiry'), (50, b'Public Metadata only (no data file access)'), (100, b'Public')])),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (249/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l154)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py [hunk 10]
```diff
@@ -0,0 +1,600 @@
...
+        ),
+        migrations.CreateModel(
+            name='License',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('name', models.CharField(unique=True, max_length=400)),
+                ('url', models.URLField(help_text=b'Link to document outlining licensing details.', unique=True, max_length=255)),
+                ('internal_description', models.TextField()),
+                ('image_url', models.URLField(max_length=255, blank=True)),
+                ('allows_distribution', models.BooleanField(default=False, help_text=b'Does this license provide distribution rights?')),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (137/100)](https://www.codacy.com/app/mytardis/mytardis/file/2683263985/issues/source?bid=2125231&fileBranchId=3335956#l279)

File: tardis/tardis_portal/migrations/0011_auto_20160505_1643.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+    dependencies = [
+        ('tardis_portal', '0010_auto_20160503_1443'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='datafile',
+            name='directory',
+            field=tardis.tardis_portal.models.fields.DirectoryField(max_length=255, null=True, blank=True),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (107/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735632/issues/source?bid=2125231&fileBranchId=3335956#l18)

File: tardis/tardis_portal/migrations/0011_auto_20160505_1643.py [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        ),
+        migrations.AlterField(
+            model_name='dataset',
+            name='directory',
+            field=tardis.tardis_portal.models.fields.DirectoryField(max_length=255, null=True, blank=True),
+        ),
+        migrations.AlterField(
+            model_name='experimentauthor',
+            name='url',
+            field=models.URLField(help_text=b'URL identifier for the author', max_length=255, null=True, blank=True),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (117/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735632/issues/source?bid=2125231&fileBranchId=3335956#l28)

File: tardis/tardis_portal/migrations/0011_auto_20160505_1643.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        ),
+        migrations.AlterField(
+            model_name='license',
+            name='name',
+            field=models.CharField(unique=True, max_length=255),
+        ),
+        migrations.AlterField(
+            model_name='license',
+            name='url',
+            field=models.URLField(help_text=b'Link to document outlining licensing details.', max_length=255),
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (110/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735632/issues/source?bid=2125231&fileBranchId=3335956#l43)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 1 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (590/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l33)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 2 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (232/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l313)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 3 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (189/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l534)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 4 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (107/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l568)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 5 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (128/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l280)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 7 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (101/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l81)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 6 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (114/100)](https://www.codacy.com/app/mytardis/mytardis/file/2736735622/issues/source?bid=2125231&fileBranchId=3335956#l45)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 1 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (590/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l33)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 2 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (232/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l313)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 3 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (189/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l534)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 4 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (107/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l568)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 5 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (128/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l280)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 7 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (101/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l81)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 6 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (114/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l45)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 8 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (122/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l296)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 9 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (249/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l154)

File: tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py (hunk 10 above)
Comment (grischa) [inline tardis/tardis_portal/migrations/0001_squashed_0011_auto_20160505_1643.py]: Codacy Issue found: [Line too long (137/100)](https://www.codacy.com/app/mytardis/mytardis/file/2752554074/issues/source?bid=2125231&fileBranchId=3335956#l279)

**Comment by jasonrig** [comment]
Merged as part of PR 653 👍

**Pull request closed without merging**